platform/sarif.go

```diff
@@ -265,3 +287,110 @@ func ReportId(projectName string) string {
 func JobUrl() string {
 	return os.Getenv("QODANA_JOB_URL")
 }
+
+func getRuleDescription(report *sarif.Report, ruleId string) string {
+	for _, run := range report.Runs {
+		for _, extension := range run.Tool.Extensions {
+			for _, rule := range extension.Rules {
+				if rule.Id == ruleId {
+					return fmt.Sprintf("Short description: %v\nFull description: %v",
+						rule.ShortDescription.Text,
+						rule.FullDescription.Text)
+				}
+			}
+		}
+	}
+	return ""
+}
+
+// ProcessSarif concludes the result of analysis based on provided SARIF file
+// - can print problems to the output
+// - can create GitLab CodeQuality issues report
+// - can submit problems to BitBucket Code Insights
+func ProcessSarif(sarifPath, analysisId, reportUrl string, printProblems, codeClimate, codeInsights bool) {
+	newProblems := 0
+	s, err := ReadReport(sarifPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var codeClimateIssues []CCIssue
+	var codeInsightIssues []bbapi.ReportAnnotation
+	if printProblems {
+		EmptyMessage()
+	}
+	for _, run := range s.Runs {
+		for _, r := range run.Results {
+			ruleId := r.RuleId
+			message := r.Message.Text
+			baselineState := baselineStateEmpty
+			if r.BaselineState != nil {
+				baselineState = r.BaselineState.(string)
+			}
+			if baselineState == baselineStateNew || baselineState == baselineStateEmpty {
+				newProblems++
+			}
+			if len(r.Locations) > 0 && baselineState != baselineStateUnchanged {
+				if codeClimate {
+					codeClimateIssues = append(codeClimateIssues, sarifResultToCodeClimate(&r))
+				}
+				if codeInsights {
+					ruleDescription := getRuleDescription(s, ruleId)
```

Comment:
Let's add simple memoization for this. Else we would keep looking up the same descriptions repeatedly, making this operation needlessly time consuming